src/spec.c

```diff
@@ -444,6 +464,21 @@ static int parseVectorField_hnsw(FieldSpec *fs, ArgsCursor *ac, QueryError *stat
     return 0;
   }
   fs->vectorOpts.expBlobSize = fs->vectorOpts.vecSimParams.hnswParams.dim * VecSimType_sizeof(fs->vectorOpts.vecSimParams.hnswParams.type);
+  size_t maxBlockSize = BLOCK_MEMORY_LIMIT / VecSimIndex_EstimateElementSize(&fs->vectorOpts.vecSimParams);
+  if (fs->vectorOpts.vecSimParams.hnswParams.blockSize == 0) {
+    if (maxBlockSize < DEFAULT_BLOCK_SIZE ) {
+      fs->vectorOpts.vecSimParams.hnswParams.blockSize = maxBlockSize;
+    } else {
+      fs->vectorOpts.vecSimParams.hnswParams.blockSize = DEFAULT_BLOCK_SIZE;
+    }
+  }
+  if (VecSimIndex_EstimateInitialSize(&fs->vectorOpts.vecSimParams) > memoryLimit - used_memory) {
+    QueryError_SetErrorFmt(status, QUERY_ELIMIT, "Vector index size exceeded server limit (%zu Bytes)", memoryLimit);
+    return 0;
+  } else if (fs->vectorOpts.vecSimParams.hnswParams.blockSize  > maxBlockSize) {
+    QueryError_SetErrorFmt(status, QUERY_ELIMIT, "Vector index block size exceeded server limit (%zu)", maxBlockSize);
```

Comment:
I don't think this error is relevant to the current state of HNSW index creation.

Reply:
right. saving time for future work